server/sonar-server/src/main/java/org/sonar/server/webhook/ws/WebhookDeliveryAction.java

```diff
@@ -0,0 +1,112 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.webhook.ws;
+
+import com.google.common.io.Resources;
+import java.util.Optional;
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.Response;
+import org.sonar.api.server.ws.WebService;
+import org.sonar.api.web.UserRole;
+import org.sonar.core.util.Uuids;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.component.ComponentDto;
+import org.sonar.db.webhook.WebhookDeliveryDto;
+import org.sonar.server.component.ComponentFinder;
+import org.sonar.server.user.UserSession;
+import org.sonarqube.ws.Webhooks;
+
+import static java.util.Objects.requireNonNull;
+import static org.sonar.server.webhook.ws.WebhookWsSupport.copyDtoToProtobuf;
+import static org.sonar.server.ws.WsUtils.checkFoundWithOptional;
+import static org.sonar.server.ws.WsUtils.writeProtobuf;
+
+public class WebhookDeliveryAction implements WebhooksWsAction {
+
+  private static final String PARAM_ID = "deliveryId";
+
+  private final DbClient dbClient;
+  private final UserSession userSession;
+  private final ComponentFinder componentFinder;
+
+  public WebhookDeliveryAction(DbClient dbClient, UserSession userSession, ComponentFinder componentFinder) {
+    this.dbClient = dbClient;
+    this.userSession = userSession;
+    this.componentFinder = componentFinder;
+  }
+
+  @Override
+  public void define(WebService.NewController controller) {
+    WebService.NewAction action = controller.createAction("delivery")
+      .setSince("6.2")
+      .setDescription("Get a webhook delivery by its id.<br/>" +
+        "Require 'Administer System' permission.<br/>" +
+        "Note that additional information are returned by api/webhooks/delivery.")
+      .setResponseExample(Resources.getResource(this.getClass(), "example-delivery.json"))
+      .setHandler(this);
+
+    action.createParam(PARAM_ID)
+      .setDescription("Id of delivery")
+      .setExampleValue(Uuids.UUID_EXAMPLE_06);
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    // fail-fast if not logged in
+    userSession.checkLoggedIn();
+
+    Data data = loadFromDatabase(request.mandatoryParam(PARAM_ID));
+    data.ensureAdminPermission(userSession);
+    data.writeTo(request, response);
+  }
+
+  private Data loadFromDatabase(String deliveryUuid) {
+    try (DbSession dbSession = dbClient.openSession(false)) {
+      Optional<WebhookDeliveryDto> delivery = dbClient.webhookDeliveryDao().selectByUuid(dbSession, deliveryUuid);
+      checkFoundWithOptional(delivery, "Webhook delivery not found");
```

Comment:
I realize this method is poorly named. It could simply be `checkFound(Optional<T>)` to be an override of `checkFound(T)`